app/controllers/measurements_controller.rb

```diff
@@ -61,7 +66,18 @@ def set_measurement
 
   # Never trust parameters from the scary internet, only allow the white list through.
   def measurement_params
-    params.require(:measurement).permit(:temperature, :sensor_id, custom_attributes: {})
+    params.require(:measurement).permit(:temperature, :sensor_id, :created_at, custom_attributes: {})
+  end
+
+  def valid_created_at?
+    return true if params.dig(:measurement, :created_at).blank?
+
+    submitted_time = Time.zone.parse(params[:measurement][:created_at])
+    return false if submitted_time.nil?
+
+    sixty_minutes_ago = 60.minutes.ago
+
+    submitted_time.between?(sixty_minutes_ago, 3.minutes.from_now)
```

Comment:
Usually validations happen in the models and a controller will just return the error if validations fail.

Reply:
Isn't it semantically correct to do it in the controller? A measurement with a date in the past isn't invalid per se, it's just invalid when submitted to that API endpoint at a certain time.